### PR TITLE
Better updates

### PR DESCRIPTION
**Pull request by BenHSmith13**
This fixes deleting, toggles reOrdering, and hides the re-arrange options if you can't re-order. Re-ordering should work, but the back-end is not using the order field like it should. waiting to hear back from cole on that one.

### CODE REVIEW

File: client/js/_author/components/assessments/question_types/multiple_choice.jsx [hunk 1]
```diff
@@ -18,27 +18,29 @@ export default class MultipleChoice extends React.Component {
...
+      this.props.updateItem({
+        question: {
+          choices: this.markedForDeletion(choice)
+        }
+      });
+    }
   }
 
-  moveChoice(choice, index, up) {
+  moveChoice(choice, up) {
```
Comment (nick-benoit14) [inline client/js/_author/components/assessments/question_types/multiple_choice.jsx]: This seems like it should be accomplished via an action, and then this logic would be in the reducer.

File: client/js/_author/components/assessments/question_types/multiple_choice_option.jsx [hunk 2]
```diff
@@ -34,7 +34,7 @@ export default function multipleChoiceOptions(props) {
 
         <div className="c-answer__icons">
           {
-            props.first ? <div className="c-answer__icons__spacer" /> : <button
+            props.first || props.shuffle ? <div className="c-answer__icons__spacer" /> : <button
```
Comment (nick-benoit14) [inline client/js/_author/components/assessments/question_types/multiple_choice_option.jsx]: Why is this happening?

File: client/js/middleware/qbank.js [hunk 3]
```diff
@@ -220,10 +221,18 @@ const qbank = {
       if (!_.isEmpty(answers)) {
         newItem.answers = answers;
       }
-      if (!_.isEmpty(choices) || item.questionString) {
+      if (!_.isEmpty(choices)
+        || updatedItem.questionString
+        || updatedItem.question.maintainOrder
```
Comment (nick-benoit14) [inline client/js/middleware/qbank.js]: Do we need both of these checks on `maintainOrder`?

File: client/js/middleware/qbank.js [hunk 4]
```diff
@@ -220,10 +221,18 @@ const qbank = {
...
+        || updatedItem.question.maintainOrder === false) {
         newItem.question = {};
-        if (item.questionString) { newItem.question.questionString = item.questionString; }
+        if (updatedItem.questionString) {
+          newItem.question.questionString = updatedItem.questionString;
+        }
         if (!_.isEmpty(choices)) { newItem.question.choices = choices; }
+        if (updatedItem.question.maintainOrder || updatedItem.question.maintainOrder === false) {
```
Comment (nick-benoit14) [inline client/js/middleware/qbank.js]: are you checking if `maintainOrder` is either not null or false? Could this be replaced with a single check to see if it is not null?

File: client/js/_author/components/assessments/question_types/multiple_choice_option.jsx (hunk 2 above)
Comment (BenHSmith13) [inline client/js/_author/components/assessments/question_types/multiple_choice_option.jsx]: good reason

File: client/js/middleware/qbank.js (hunk 3 above)
Comment (BenHSmith13) [inline client/js/middleware/qbank.js]: changing to _.isUndefined

File: client/js/_author/components/assessments/question_types/multiple_choice.jsx (hunk 1 above)
Comment (BenHSmith13) [inline client/js/_author/components/assessments/question_types/multiple_choice.jsx]: I don't disagree, but we should talk about design and change it outside the scope of this PR

**Pull request merged**